client/experiment-dispatcher/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { CurrentUser } from '@automattic/calypso-analytics/dist/types/utils/current-user';
+import { AnyAction } from 'redux';
+import { loadExperimentAssignment } from 'calypso/lib/explat';
+import { initializeCurrentUser } from 'calypso/lib/user/shared-utils';
+import '@automattic/calypso-polyfills';
+import { createReduxStore } from 'calypso/state';
+import { setCurrentUser } from 'calypso/state/current-user/actions';
+import { getInitialState, getStateFromCache } from 'calypso/state/initial-state';
+import initialReducer from 'calypso/state/reducer';
+import { setStore } from 'calypso/state/redux-store';
+import { redirectWithParamsAndLocale, getManifestFromUrl } from './utils';
+import './style.scss';
+
+window.AppBoot = async () => {
+	/**
+	 * Initialize the current user and the redux store. Some experiments may require the current user.
+	 */
+	const user = ( await initializeCurrentUser() ) as CurrentUser | false;
+	if ( user ) {
+		const userId = ( user as CurrentUser ).ID;
+
+		const initialState = getInitialState( initialReducer, userId );
+		const reduxStore = createReduxStore( initialState, initialReducer );
+		setStore( reduxStore, getStateFromCache( userId ) );
+
+		user && reduxStore.dispatch( setCurrentUser( user ) as unknown as AnyAction );
+	}
+
+	const experimentManifest = getManifestFromUrl();
+
+	try {
+		const assignment = await loadExperimentAssignment( experimentManifest.experiment_explat_name );
+
```

Comment:
@alshakero It would be nice to have a Tracks event fire around here (eg `calypso_experiment_dispatcher_run/start` or something) that we could use for:

- In the future as exposure event (I think in our case Isa is still using `calypso_signup_complete` for QA purposes)
- Even when it's not used as exposure event we can use it for our manual analyses
- In Calypso the use of `window.location.replace` in (`redirectWithParamsAndLocale`) will truncate part of the HTTP Referrer (due to Calypso's `Referrer Policy: origin` thing) so having this Tracks event would allow us to automatically record the original referrer who sent the user here - again, for analysis/QA purposes
- We could include handy metadata about the experiment too as props

Reply:
@jessie-ross if you're referring to `wpcom_experiment_variation_assigned` this does not track location nor referrer, also logstash is not an option to analyse user behaviour, attributes, issues outside of logging expected runtime exceptions (which is not what we're talking about here).

Reply:
@aaronyan Perhaps we should add more information into this event in the backend.

Reply:
@MicBosi, can you help p2 on betterexperiments more about the eventprops you wish to capture? We can consider using the backend variation assigned event to capture more details, as @jessie-ross recommends.